abci/example/kvstore/kvstore.go

```diff
@@ -432,14 +432,21 @@ func (app *Application) execPrepareTx(tx []byte) *types.ExecTxResult {
 
 // substPrepareTx subst all the preparetx in the blockdata
 // to null string(could be any arbitrary string).
-func (app *Application) substPrepareTx(blockData [][]byte) [][]byte {
-	// TODO: this mechanism will change with the current spec of PrepareProposal
-	// We now have a special type for marking a tx as changed
+func (app *Application) substPrepareTx(blockData [][]byte) []*types.TxRecord {
+	trs := make([]*types.TxRecord, len(blockData))
 	for i, tx := range blockData {
 		if isPrepareTx(tx) {
-			blockData[i] = make([]byte, len(tx))
+			trs[i] = &types.TxRecord{
+				Tx:     make([]byte, len(tx)),
```

Comment:
Do we need to copy some data into this?

Reply:
I'm realizing that this fails the `ResponsePrepareProposal` logic because it results in duplicate transactions. I'm going to instead perform the substitution by stripping off the prefix.